pyccel/codegen/printing/fcode.py

```diff
@@ -1787,8 +1787,23 @@ def _print_AugAssign(self, expr):
 
     def _print_PythonRange(self, expr):
         start = self._print(expr.start)
-        stop  = self._print(expr.stop) + '-' + self._print(LiteralInteger(1))
         step  = self._print(expr.step)
+
+        test_step = expr.step
+        if isinstance(test_step, PyccelUnarySub):
+            test_step = expr.step.args[0]
+
+        # testing if the step is a value or an expression
+        if isinstance(test_step, Literal):
+            stop = PyccelMinus(expr.stop, LiteralInteger(1))
+            if isinstance(expr.step, PyccelUnarySub):
+                stop = PyccelAdd(expr.stop, LiteralInteger(1))
```

Comment:
See above, why not use an if/else rather than setting stop twice?

Reply:
Here too is fixed in [cec0ecc](https://github.com/pyccel/pyccel/pull/780/commits/cec0ecc4ab3d036b396f014715bbe158e3f12a44)